### PR TITLE
Add missing icons + case-insensitive support

### LINKED ISSUE #3: Add missing icons
## Problem description
These should be added:
* `html` *HTML*
* `php` *PHP*
* `jsm` *JavaScript Module*
* `njsproj` *Visual Studio project file*

And can you make it **case-insensitive**?

### PR DESCRIPTION
**Pull request by EntityPlantt**
## What's changed
* Added icon for `php`
* Added `html`, `jsm` and `njsproj` to the list

### CODE REVIEW

**Comment by minhazmiraz** [comment]
still not working. You have to add icon name in **materialFileIconsName.js**.
And change the php icon to this:

> php:
>     '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">\n<path d="M12 18.08c-6.63 0-12-2.72-12-6.08s5.37-6.08 12-6.08S24 8.64 24 12s-5.37 6.08-12 6.08m-5.19-7.95c.54 0 .91.1 1.09.31.18.2.22.56.13 1.03-.1.53-.29.87-.58 1.09-.28.22-.71.33-1.29.33h-.87l.53-2.76h.99m-3.5 5.55h1.44l.34-1.75h1.23c.54 0 .98-.06 1.33-.17.35-.12.67-.31.96-.58.24-.22.43-.46.58-.73.15-.26.26-.56.31-.88.16-.78.05-1.39-.33-1.82-.39-.44-.99-.65-1.82-.65H4.59l-1.28 6.58m7.25-8.33-1.28 6.58h1.42l.74-3.77h1.14c.36 0 .6.06.71.18.11.12.13.34.07.66l-.57 2.93h1.45l.59-3.07c.13-.62.03-1.07-.27-1.36-.3-.27-.85-.4-1.65-.4h-1.27L12 7.35h-1.44M18 10.13c.55 0 .91.1 1.09.31.18.2.22.56.13 1.03-.1.53-.29.87-.57 1.09-.29.22-.72.33-1.3.33h-.85l.5-2.76h1m-3.5 5.55h1.44l.34-1.75h1.22c.55 0 1-.06 1.35-.17.35-.12.65-.31.95-.58.24-.22.44-.46.58-.73.15-.26.26-.56.32-.88.15-.78.04-1.39-.34-1.82-.36-.44-.99-.65-1.82-.65h-2.75l-1.29 6.58z" fill="#1E88E5"/>\n</svg>'

File: src/components/icons/material/materialFileIcons6.js [hunk 1]
```diff
@@ -131,4 +131,6 @@ export const materialFileIcons6 = {
...
...614 39.6,-13.758 37.91,-22.466 36.012,-32.217 32.493,-39.13 27.452,-43.012 22.29,-46.986 13.898,-49 2.511,-49 L -9.522,-49 0,0 Z m 28.831,35 -61.054,0 c -2.872,0 -5.341,-2.036 -5.889,-4.855 L -66.44,-115.606 c -0.342,-1.759 0.12,-3.578 1.259,-4.961 1.14,-1.383 2.838,-2.183 4.63,-2.183 l 31.75,0 c 2.872,0 5.342,2.036 5.89,4.855 l 6.587,33.895 22.249,0 c 12.582,0 23.174,1.372 31.479,4.077 8.541,2.775 16.401,7.481 23.356,13.986 5.752,5.291 10.488,11.23 14.078,17.655 3.591,6.427 6.171,13.594 7.668,21.302 3.715,19.105 0.697,34.403 -8.969,45.467 C 63.965,29.444 48.924,35 28.831,35 m -45.632,-90 19.312,0 c 12.801,0 22.336,2.411 28.601,7.234 6.267,4.824 10.492,12.875 12.688,24.157 2.102,10.832 1.145,18.476 -2.871,22.929 C 36.909,3.773 28.87,6 16.808,6 L -4.946,6 -16.801,-55 M 28.831,29 C 47.198,29 60.597,24.18 69.019,14.539 77.441,4.898 79.976,-8.559 76.616,-25.836 75.233,-32.953 72.894,-39.46 69.601,-45.355 66.304,-51.254 61.999,-56.648 56.679,-61.539 50.339,-67.472 43.296,-71.7 35.546,-74.218 27.796,-76.743 17.925,-78 5.925,-78 l -27.196,0 -7.53,-38.75 -31.75,0 28.328,145.75 61.054,0" id="path3506" style="fill:#ffffff;fill-opacity:1;fill-rule:nonzero;stroke:none"/></g></g></g></g></svg>'
```
Comment (minhazmiraz) [inline src/components/icons/material/materialFileIcons6.js]: Change the icon to this:
`<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">\n<path d="M12 18.08c-6.63 0-12-2.72-12-6.08s5.37-6.08 12-6.08S24 8.64 24 12s-5.37 6.08-12 6.08m-5.19-7.95c.54 0 .91.1 1.09.31.18.2.22.56.13 1.03-.1.53-.29.87-.58 1.09-.28.22-.71.33-1.29.33h-.87l.53-2.76h.99m-3.5 5.55h1.44l.34-1.75h1.23c.54 0 .98-.06 1.33-.17.35-.12.67-.31.96-.58.24-.22.43-.46.58-.73.15-.26.26-.56.31-.88.16-.78.05-1.39-.33-1.82-.39-.44-.99-.65-1.82-.65H4.59l-1.28 6.58m7.25-8.33-1.28 6.58h1.42l.74-3.77h1.14c.36 0 .6.06.71.18.11.12.13.34.07.66l-.57 2.93h1.45l.59-3.07c.13-.62.03-1.07-.27-1.36-.3-.27-.85-.4-1.65-.4h-1.27L12 7.35h-1.44M18 10.13c.55 0 .91.1 1.09.31.18.2.22.56.13 1.03-.1.53-.29.87-.57 1.09-.29.22-.72.33-1.3.33h-.85l.5-2.76h1m-3.5 5.55h1.44l.34-1.75h1.22c.55 0 1-.06 1.35-.17.35-.12.65-.31.95-.58.24-.22.44-.46.58-.73.15-.26.26-.56.32-.88.15-.78.04-1.39-.34-1.82-.36-.44-.99-.65-1.82-.65h-2.75l-1.29 6.58z" fill="#1E88E5"/>\n</svg>`


Also add the file name in **materialFileIconsName.js**.

File: src/components/api/MaterialIconApi.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { materialFolderNamesToIcons } from "../gen/materialFolderIconsName";
 import Icons from "../icons/index";
 
 export const getMaterialFileIcon = (fileName) => {
+	fileName = fileName.toLowerCase();
```
Comment (minhazmiraz) [inline src/components/api/MaterialIconApi.js:10]: Add this for VS icons too.

**Comment by minhazmiraz** [comment]
Please try to test it before submitting the PR. Thank you 😊

File: src/components/icons/material/materialFileIcons6.js (hunk 1 above)
Comment (EntityPlantt) [reply src/components/icons/material/materialFileIcons6.js]: Updated: https://github.com/EntityPlantt/file-extension-icon-JS/commit/02360cb08fa2d7020dbe5656c5cdfba8efeb42e8

File: src/components/api/MaterialIconApi.js (hunk 2 above)
Comment (EntityPlantt) [reply src/components/api/MaterialIconApi.js:10]: Added in https://github.com/minhazmiraz/file-extension-icon-JS/pull/4/commits/1c8b603de41b956be4b3b306b8e9e7363c58eb8a

**Comment by EntityPlantt** [comment]
> **@minhazmiraz wrote:**
> Please try to test it before submitting the PR. Thank you 😊

Sorry, but I don't know how to build the package. (I've never built any NPM packages.)  
Can you reply with a step-to-step building process?

**Comment by minhazmiraz** [comment]
> > **@minhazmiraz wrote:**
> > Please try to test it before submitting the PR. Thank you 😊
> 
> Sorry, but I don't know how to build the package. (I've never built any NPM packages.) Can you reply with a step-to-step building process?

1. Pull the code from github
2. run **npm install** in console
3. run **npm run start** to run the code in localhost

**Comment by EntityPlantt** [comment]
> **@minhazmiraz wrote:**
> > > **@minhazmiraz wrote:**
> > > Please try to test it before submitting the PR. Thank you 😊
> > 
> > 
> > Sorry, but I don't know how to build the package. (I've never built any NPM packages.) Can you reply with a step-to-step building process?
> 
> 1. Pull the code from github
> 2. run **npm install** in console
> 3. run **npm run** to run the code in localhost

I get this error when running `npm start`:
```plain
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/tokenize' is not defined by "exports" in C:\Users\NikTea\Documents\GitHub\file-extension-icon-JS\node_modules\postcss-safe-parser\node_modules\postcss\package.json
    at new NodeError (node:internal/errors:371:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:429:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:683:3)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\NikTea\Documents\GitHub\file-extension-icon-JS\node_modules\postcss-safe-parser\lib\safe-parser.js:1:17) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```
I've searched on the Internet, but it shows [0 results](https://www.google.com/search?q=Error+%5BERR_PACKAGE_PATH_NOT_EXPORTED%5D%3A+Package+subpath+%27.%2Flib%2Ftokenize%27+is+not+defined+by+%22exports%22+in+C%3A%5CUsers%5CNikTea%5CDocuments%5CGitHub%5Cfile-extension-icon-JS%5Cnode_modules%5Cpostcss-safe-parser%5Cnode_modules%5Cpostcss%5Cpackage.json&rlz=1C1LOQA_enMK915MK915&sourceid=chrome&ie=UTF-8).  
What can I do about it?

Tried running `npm test`, it shows a testing dialog.  
Also tried `npm run`, `npm run build` and `num run eject`. None of them works.

**Comment by EntityPlantt** [comment]
And also one thing, while I'm at it: Change this **folder** word to **file**

image

**Comment by minhazmiraz** [comment]
> > > **@minhazmiraz wrote:**
> > > Please try to test it before submitting the PR. Thank you 😊
> > 
> > 
> > Sorry, but I don't know how to build the package. (I've never built any NPM packages.) Can you reply with a step-to-step building process?
> 
> 1. Pull the code from github
> 2. run **npm install** in console
> 3. run **npm run start** to run the code in localhost

Sorry, it should be **npm run start**.

**Pull request merged**